### ISSUE TITLE
Headers Applied to the WebClientBuilder are Causing Issues, But Not When Applied to the ClientRequest Afterwards

### ISSUE THREAD

**Issue by CamdenSchiman** [helidon-io/helidon#8881]
## Environment Details
* Helidon Version: 4.0.7
* Helidon MP
* JDK version: 21.0.2
* OS: MacOS Sonoma 14.5
* Docker version (if applicable):

----------

## Problem Description
I am using Helidon's WebClient to make a series of API calls. The API calls usually start with a call to generate an auth token. Then, the auth token is used in subsequent calls. This process was working fine when using Helidon 3, but after upgrading to Helidon 4, I am getting a 400 Bad Request error. When recreating these calls in Postman and using cURL, they execute fine. The issue seems to be with how I'm setting the headers. This was my previous method of setting the headers:
```
webClientBuilder.addHeader("customHeader1", value1).addHeader("customHeader2", value2).addHeader(HeaderNames.AUTHORIZATION, authToken);
webClientBuilder.build().get();
```
Here, when printing the headers, I noticed the AUTHORIZATION has a different format that the other headers, even when I use the string `"Authorization"` rather than `HeaderNames.AUTHORIZATION`. However, if I set the header this way, it works:
```
webClientBuilder.addHeader("customHeader1", value1).addHeader("customHeader2", value2);
webClientBuilder.build().get().header(HeaderNames.AUTHORIZATION, authToken);
```
I believe the other headers are still being processed correctly, but I have no sure way to check. This seems like an issue with the WebClientBuilder's addHeader method when it encounters a HeaderNames value, or a string equivalent to one of the HeaderNames values.

## Steps to reproduce
Steps for reproducing and example code are noted above.

**Comment by tomas-langer** [comment]
Hello, there was a bug in 4.0.7. that duplicated headers registered with client builder.
This is now fixed. Could you please try with 4.0.10?
I have reproduced the problem - there is a duplicated header with 4.0.7, and just a single one as expected with 4.0.10 (including `Authorization` header).

**Comment by CamdenSchiman** [comment]
Yes, it looks like this issue was resolved with the upgrade from 4.0.7 to 4.0.10. Thank you.

From: Tomas Langer ***@***.***>
Date: Thursday, June 13, 2024 at 6:39 AM
To: helidon-io/helidon ***@***.***>
Cc: Camden Schiman ***@***.***>, Author ***@***.***>
Subject: [External] : Re: [helidon-io/helidon] Headers Applied to the WebClientBuilder are Causing Issues, But Not When Applied to the ClientRequest Afterwards (Issue #8881)

Hello, there was a bug in 4.0.7. that duplicated headers registered with client builder.
This is now fixed. Could you please try with 4.0.10?
I have reproduced the problem - there is a duplicated header with 4.0.7, and just a single one as expected with 4.0.10 (including Authorization header).
